### PR TITLE
Unable to create submission journals if approval_hierarchy is installed

### PR DESCRIPTION
**Pull request by DaeronAlagos**
If the approval_hierarchy module is installed as well as the mtd module, when the VAT is submitted, the current user may not have permission to post a move.line for the VAT submission values.

Fixes #16990

### CODE REVIEW

File: account_mtd_vat/models/mtd_vat_issue_request.py
Comment (Nick-OpusVL) [inline account_mtd_vat/models/mtd_vat_issue_request.py:560]:
```suggestion
        account_move_id.sudo().action_post()
```

File: account_mtd_vat/models/mtd_vat_issue_request.py
Comment (Nick-OpusVL) [inline account_mtd_vat/models/mtd_vat_issue_request.py:10]:
```suggestion
from odoo import models, fields, api, exceptions
```

**Review by Nick-OpusVL — CHANGES_REQUESTED** [review]
OK mostly, but with_user is not intended for bypassing access rights checks - use sudo() for that unless somehow it doesn't work in this particular case.

https://www.odoo.com/documentation/13.0/reference/orm.html#altering-the-environment

The difference with sudo is it doesn't actually change the user, so things like audit trail messages, From addresses for emails (if any), default company etc won't be affected by it.

Please read the warnings about sudo() in the documentation link above though.

**Comment by Nick-OpusVL** [comment]
@DaeronAlagos please heed the warnings about sudo() in the link I posted you though.

**Comment by DaeronAlagos** [comment]
Good spot. Now I need to go and check what method is being used in approval_hierarchy to check the user permissions. I'll re-request once I've made the correct change.

**Comment by DaeronAlagos** [comment]
I think you're correct about this to be using .sudo() instead of .with_user() in a normal scenario. Unfortunately approval_hierarchy is using a custom method to check the user (id==SUPERUSER_ID or login='admin'). I'm not going to change it to .sudo() here as that will require greater changes to approval_hierarchy.

**Comment by DaeronAlagos** [comment]
I'm not going to merge this once approved. I will update RM submodule to point to this commit so it can be tested on staging before being merged and deployed to production.

**Review by Nick-OpusVL — APPROVED** [review]
OK then.

**Pull request merged**
